contracts/Registry.sol

```diff
@@ -94,6 +125,52 @@ contract Registry is IRegistry {
         );
     }
 
+    /**
+     * @dev Alter a DSNP Id resolution address by EIP-712 Signature
+     * @param v EIP-155 calculated Signature v value
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param change Change data containing nonce, new address and handle
+     */
+    function changeAddressByEIP712Sig(
+        uint8 v,
+        bytes32 r,
+        bytes32 s,
+        AddressChange calldata change
+    ) external override {
+        // Checks
+
+        Registration storage reg = registrations[change.handle];
+        require(reg.id != 0, "Handle does not exist");
+        require(reg.nonce == change.nonce, "Nonces do not match");
+
+        address signer = addressChangeSigner(v, r, s, change);
+
+        // Effects
+
+        reg.nonce++;
+
+        address oldAddr = reg.identityAddress;
+        reg.identityAddress = change.addr;
+        emit DSNPRegistryUpdate(reg.id, change.addr, change.handle);
+
+        // Interactions
+
+        // ensure old delegation contract authorizes this change
+        IDelegation oldAuth = IDelegation(oldAddr);
```

Comment:
should we check the permissions around ownership transfer before we emit  the registry update?